pkg/webhook/validate.go

```diff
@@ -57,6 +63,21 @@ func staticValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePol
 		return false, fmt.Errorf("at least one of these parameters (Vendor, DeviceID or PfNames) has to be defined in nicSelector in CR %s", cr.GetName())
 	}
 
+	if cr.Spec.NicSelector.Vendor != "" {
+		if !sriovnetworkv1.StringInArray(cr.Spec.NicSelector.Vendor, keys(SupportedModels)) {
+			return false, fmt.Errorf("vendor %s is not supported", cr.Spec.NicSelector.Vendor)
+		}
+		if cr.Spec.NicSelector.DeviceID != "" {
+			if (cr.Spec.NicSelector.Vendor == IntelID && !sriovnetworkv1.StringInArray(cr.Spec.NicSelector.DeviceID, SupportedModels[IntelID])) || (cr.Spec.NicSelector.Vendor == MellanoxID && !sriovnetworkv1.StringInArray(cr.Spec.NicSelector.DeviceID, SupportedModels[MellanoxID])) {
+				return false, fmt.Errorf("vendor/device %s/%s is not supported", cr.Spec.NicSelector.Vendor, cr.Spec.NicSelector.DeviceID)
+			}
+		}
+	} else if cr.Spec.NicSelector.DeviceID != "" {
+		if !sriovnetworkv1.StringInArray(cr.Spec.NicSelector.DeviceID, append(SupportedModels[IntelID], SupportedModels[MellanoxID]...)) {
+			return false, fmt.Errorf("device %s is not supported", cr.Spec.NicSelector.DeviceID)
+		}
+	}
+
```

Comment:
I think we still allow use of other type of NICs if vendor/device are not explicitly specified. Shall we change the behavior as well?

Reply:
Users can still do that by disabling the operator webhook.

Reply:
Right, but I think it's different issue when disabling webhook where we have no control over which IDs are configured.

Reply:
The implication is that, when users choose to disable the webhook, they shall know exactly what they are trying to do. Otherwise, it shall not be disabled.

Also, I agree to @dankenigsberg 's comments that we shall not have the validation logic at more than 1 place.